prog/dftb+/lib_dftbplus/parser.F90

```diff
@@ -3974,9 +3974,108 @@ subroutine readEeqModel(node, input, geo, nrChrg, kChiDefault, kGamDefault, &
     call getChildValue(node, "EwaldParameter", input%parEwald, 0.0_dp)
     call getChildValue(node, "EwaldTolerance", input%tolEwald, 1.0e-9_dp)
 
+    call readCoordinationNumber(node, input%cnInput, geo, "erf", 8.0_dp)
+
   end subroutine readEeqModel
 
 
+  !> Read in coordination number settings
+  subroutine readCoordinationNumber(node, input, geo, cnDefault, cutDefault)
+
+    !> Node to get the information from
+    type(fnode), pointer :: node
+
+    !> Control structure to be filled
+    type(TCNInput), intent(inout) :: input
+
+    !> Geometry structure to be filled
+    type(TGeometry), intent(in) :: geo
+
+    !> Default value for the coordination number type
+    character(len=*), intent(in) :: cnDefault
+
+    !> Default value for the maximum CN used for cutting (0 turns it off)
+    real(dp), intent(in) :: cutDefault
+
+    type(fnode), pointer :: value1, value2, child, child2, field
+    type(string) :: buffer, modifier
+    real(dp), allocatable :: kENDefault(:), kRadDefault(:)
+    integer :: iSp
+
+    call getChildValue(node, "CoordinationNumber", value1, cnDefault, child=child)
+    call getNodeName(value1, buffer)
+
+    select case(char(buffer))
+    case default
+      call detailedError(child, "Invalid coordination number type specified")
+  #:for cnType in ("exp", "erf", "cov", "gfn")
```

Comment:
Please do not to use pre-processor loop here. You only spare 4 lines of code all together, it is not worth of the obfuscation.

Also, I'd find it more logical to have the `case default` clause at the end (as it would be in an `if / else if / ... / else` construct)

Reply:
I prefer to have the default on-top because it usually contains the error handling, which I find easier to understand when it is close to the failing condition. I also negate most of my `if` constructs to reflect this behaviour, so the if/then/else is not really an argument, at least for me.

Reply:
Makes sense, I can live with that logic as well. :wink: